dbms/src/Flash/Planner/Plans/PhysicalJoin.cpp

```diff
@@ -324,14 +323,58 @@ void PhysicalJoin::buildPipeline(PipelineBuilder & builder, Context & context, P
     builder.addPlanNode(join_probe);
 }
 
-void PhysicalJoin::finalize(const Names & parent_require)
+void PhysicalJoin::finalizeImpl(const Names & parent_require)
 {
-    // schema.size() >= parent_require.size()
     FinalizeHelper::checkSchemaContainsParentRequire(schema, parent_require);
+    join_ptr->finalize(parent_require);
+    auto required_input_columns = join_ptr->getRequiredColumns();
+    std::unordered_set<String> build_schema_sets;
+    for (const auto & name : build_side_prepare_actions->getSampleBlock().getNames())
+    {
+        build_schema_sets.insert(name);
+    }
+    std::unordered_set<String> probe_schema_sets;
```

Comment:
seems `set<string>` is useless, because there are no duplicate column names in the block...

Reply:
Refined.